### ISSUE TITLE
Syslog OUTPUT crashing with Mode TLS

### ISSUE THREAD

**Issue by oshelot** [fluent/fluent-bit#5643]
## Bug Report

**Describe the bug**
FLB crashes when using mode **_tls_** with SYSLOG OUTPUT. The following configuration produces an error:

```
[OUTPUT]
    name                 syslog
    match                tag_out
    host                 syslogserver.net
    port                 443
    mode                 tls
    syslog_format        rfc3164
    syslog_message_key   message
    syslog_severity_key  pri
    syslog_facility_key  facility
    syslog_hostname_key  host
    syslog_appname_key   ident
    tls.debug		 4
    tls.verify           off
    tls.ca_file          /path/chain.crt
    tls.crt_file         /path/domain.crt
    tls.key_file         /path/domain.key
    tls.key_passwd	 password (tried without password also)

```
The following config works:
```
[OUTPUT]
    name                 syslog
    match                tag_out
    host                 localhost.syslog
    port                 514
    mode                 tcp
    syslog_format        rfc3164
    syslog_message_key   message
    syslog_severity_key  pri
    syslog_facility_key  facility
    syslog_hostname_key  host
    syslog_appname_key   ident

```

**Expected behavior**
Syslog should work with TLS enabled

**Screenshots**
Error:
```
sudo /opt/fluent-bit/bin/fluent-bit -c /etc/fluent-bit/syslog-out-ssl.conf 
Fluent Bit v1.9.5
* Copyright (C) 2015-2022 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/06/26 06:33:41] [ info] Configuration:
[2022/06/26 06:33:41] [ info]  flush time     | 1.000000 seconds
[2022/06/26 06:33:41] [ info]  grace          | 5 seconds
[2022/06/26 06:33:41] [ info]  daemon         | 0
[2022/06/26 06:33:41] [ info] ___________
[2022/06/26 06:33:41] [ info]  inputs:
[2022/06/26 06:33:41] [ info]      syslog
[2022/06/26 06:33:41] [ info] ___________
[2022/06/26 06:33:41] [ info]  filters:
[2022/06/26 06:33:41] [ info]      lua.0
[2022/06/26 06:33:41] [ info] ___________
[2022/06/26 06:33:41] [ info]  outputs:
[2022/06/26 06:33:41] [ info]      syslog.0
[2022/06/26 06:33:41] [ info] ___________
[2022/06/26 06:33:41] [ info]  collectors:
[2022/06/26 06:33:43] [engine] caught signal (SIGSEGV)
#0  0x562d5e33a160      in  flb_tls_session_create() at src/tls/flb_tls.c:334
#1  0x562d5e3489d5      in  flb_io_net_connect() at src/flb_io.c:109
#2  0x562d5e3205d6      in  create_conn() at src/flb_upstream.c:560
#3  0x562d5e320b78      in  flb_upstream_conn_get() at src/flb_upstream.c:705
#4  0x562d5e40f668      in  cb_syslog_flush() at plugins/out_syslog/syslog.c:768
#5  0x562d5e303e6b      in  output_pre_cb_flush() at include/fluent-bit/flb_output.h:517
#6  0x562d5e8a6f8a      in  co_init() at lib/monkey/deps/flb_libco/amd64.c:117
#7  0xffffffffffffffff  in  ???() at ???:0
Aborted
```
Log:
```
[2022/06/26 06:33:41] [ info] [fluent bit] version=1.9.5, commit=, pid=2775
[2022/06/26 06:33:41] [debug] [engine] coroutine stack size: 24576 bytes (24.0K)
[2022/06/26 06:33:41] [ info] [storage] version=1.2.0, type=memory+filesystem, sync=normal, checksum=disabled, max_chunks_up=128
[2022/06/26 06:33:41] [ info] [storage] backlog input plugin: storage_backlog.1
[2022/06/26 06:33:41] [ info] [cmetrics] version=0.3.4
[2022/06/26 06:33:41] [debug] [syslog:syslog.0] created event channels: read=24 write=25
[2022/06/26 06:33:41] [ info] [in_syslog] TCP server binding 0.0.0.0:13010
[2022/06/26 06:33:41] [debug] [storage_backlog:storage_backlog.1] created event channels: read=27 write=28
[2022/06/26 06:33:41] [ info] [input:storage_backlog:storage_backlog.1] queue memory limit: 488.3M
[2022/06/26 06:33:41] [debug] [syslog:syslog.0] created event channels: read=20 write=29
[2022/06/26 06:33:41] [ info] [output:syslog:syslog.0] setup done for us.elb.relay.logtrust.net:443
[2022/06/26 06:33:41] [debug] [router] match rule syslog.0:syslog.0
[2022/06/26 06:33:42] [ info] [http_server] listen iface=0.0.0.0 tcp_port=2020
[2022/06/26 06:33:42] [ info] [sp] stream processor started
[2022/06/26 06:33:42] [ info] [input:storage_backlog:storage_backlog.1] register syslog.0/2313-1656224697.74443273.flb
[2022/06/26 06:33:42] [ info] [input:storage_backlog:storage_backlog.1] register syslog.0/2313-1656224697.830082371.flb
[2022/06/26 06:33:42] [ info] [input:storage_backlog:storage_backlog.1] queueing syslog.0:2313-1656224697.74443273.flb
[2022/06/26 06:33:42] [ info] [input:storage_backlog:storage_backlog.1] queueing syslog.0:2313-1656224697.830082371.flb
[2022/06/26 06:33:43] [debug] [task] created task=0x7fcc556625b0 id=0 OK
[2022/06/26 06:33:43] [debug] [task] created task=0x7fcc55662620 id=1 OK

```

**Your Environment**

* Version used: 1.9.5
* Configuration: See about. Testing syslog in and syslog out
* Environment name and version:
Fresh Ubuntu 20.04 LTS install on ESX VM
* Server type and version: ESX V6.7
* Operating System and version:
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.4 LTS
Release:	20.04
Codename:	focal
* Filters and plugins:

**Additional context**

**Comment by edsiper** [comment]
fixed by #5646

**Comment by danielzhanghl** [comment]
@edsiper could you help to confirm the config format, should that like below, there is no any document show how to use that,  like https://docs.fluentbit.io/manual/v/2.0-pre/pipeline/outputs/syslog
thanks.

[OUTPUT]
    name                 syslog
    match                tag_out
    host                 syslogserver.net
    port                 443
    mode                 tcp
    syslog_format        rfc3164
    syslog_message_key   message
    syslog_severity_key  pri
    syslog_facility_key  facility
    syslog_hostname_key  host
    syslog_appname_key   ident

[TLS]
    tls.debug		 4
    tls.verify           off
    tls.ca_file          /path/chain.crt
    tls.crt_file         /path/domain.crt
    tls.key_file         /path/domain.key
    tls.key_passwd	 password (tried without password also)
